skopt/space/space.py

```diff
@@ -550,6 +552,49 @@ def is_real(self):
         """
         return all([isinstance(dim, Real) for dim in self.dimensions])
 
+    @classmethod
+    def from_yaml(cls, yml_path, namespace=None):
+        """Create Space from yaml configuration file
+
+        Parameters
+        ----------
+        * `yml_path` [str]:
+           Full path to yaml configuration file
+        * `namespace` [str, default=None]:
+           Namespace within configuration file to use
+
+        Returns
+        -------
+        * `space` [Space]:
+           Instantiated Space object
+        """
+        with open(yml_path, 'rb') as f:
+            cfg = yaml.load(f)
+
+        this_module = sys.modules[__name__]
+        # Dynamically create list of subclass names for Dimension
+        dimension_classes = ['Real', 'Integer', 'Categorical']
+
+        # Extract space options for configuration file
+        options_name = namespace or list(cfg.keys())[0]
+        options = cfg if isinstance(cfg, list) else cfg[options_name]
+
+        # Populate list with Dimension objects
+        dimensions = []
+        for option in options:
+            key = list(option.keys())[0]
```

Comment:
this having to convert iterators into lists this feels/reads awkward but I don't have a better idea how to make it look "nicer" :-(

Reply:
haha, i was hoping i just didn't know the clean way =)